src/Manager.sol

```diff
@@ -681,4 +696,44 @@ abstract contract Manager is
     function isMessageExecuted(bytes32 digest) public view returns (bool) {
         return _getMessageAttestationsStorage()[digest].executed;
     }
+
+    function getSibling(uint16 chainId) public view returns (bytes memory) {
+        return _getSiblingsStorage()[chainId];
+    }
+
+    function setSibling(uint16 chainId, bytes memory siblingContract) external onlyOwner {
+        if (chainId == 0) {
+            revert InvalidSiblingChainIdZero();
+        }
+        if (siblingContract.length == 0) {
+            revert InvalidSiblingZeroLength();
+        }
+        if (_isAllZeros(siblingContract)) {
+            revert InvalidSiblingZeroBytes();
+        }
+        _getSiblingsStorage()[chainId] = siblingContract;
+    }
+
+    function _isAllZeros(bytes memory payload) internal pure returns (bool) {
+        for (uint256 i = 0; i < payload.length; i++) {
+            if (payload[i] != 0) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    function _areBytesEqual(bytes memory a, bytes memory b) internal pure returns (bool) {
+        if (a.length != b.length) {
+            return false;
+        }
+
+        for (uint256 i = 0; i < a.length; i++) {
+            if (a[i] != b[i]) {
+                return false;
+            }
+        }
+
+        return true;
```

Comment:
how expensive is this? is there a way to reduce the gas cost by batch loading word by word instead of byte by bytes?

Reply:
Synced offline -- we'll open a follow up PR to use the [BytesLib equal method](https://github.com/GNSPS/solidity-bytes-utils/blob/master/contracts/BytesLib.sol#L396-L503) to compare words, with the assumption that memory is word aligned (which it should be since we don't have any weird assembly blocks).

Opened an issue for this -- https://github.com/wormhole-foundation/example-native-token-transfers/issues/36